src/libs/browser-storage.js

```diff
@@ -8,6 +8,46 @@ import { maybeParseJSON, subscribable } from 'src/libs/utils'
  * values might be deleted in case of space overflow. For critical data, use the 'static' version.
  */
 
+class InMemoryStorage {
+  storage = Object.create(null)
+
+  get length() {
+    return _.size(this.storage)
+  }
+
+  key(index) {
+    return Object.keys(this.storage)[index] || null
+  }
+
+  getItem(key) {
+    return _.has(key, this.storage) ? _.get(key, this.storage) : null
+  }
+
+  setItem(key, value) {
+    this.storage[key] = value
+  }
+
+  removeItem(key) {
+    delete this.storage[key]
+  }
+
+  clear() {
+    this.storage = Object.create(null)
+  }
+}
+
+export const getStorage = storageType => {
+  if (!_.includes(storageType, ['local', 'session'])) {
+    throw new Error(`Unknown storage type: ${storageType}`)
+  }
+
+  try {
+    return window[`${storageType}Storage`]
+  } catch (error) {
+    return new InMemoryStorage()
+  }
+}
+
```

Comment:
Just because strings can draw the eye with, "do I need to worry if that is misspelled," I might also add:
```javascript
export const getLocalStorage = () => getStorage('local')
export const getSessionStorage = () => getStorage('session')
```

Reply:
Got rid of the strings entirely in https://github.com/DataBiosphere/terra-ui/pull/3391/commits/662f7f8743d9b0bfdd835cd1a8cd18e99e928b8c.